File: pocketr-java.api/src/main/java/com/decrux/pocketr/api/repositories/projections/MonthlyExpenseProjection.java
package com.decrux.pocketr.api.repositories.projections;

import java.util.UUID;

public record MonthlyExpenseProjection(
    UUID expenseAccountId,
    String expenseAccountName,
    UUID categoryTagId,
    String categoryTagName,
    String currency,
    long netMinor
) {
    public UUID getExpenseAccountId() {
        return expenseAccountId;
    }

    public String getExpenseAccountName() {
        return expenseAccountName;
    }

    public UUID getCategoryTagId() {
        return categoryTagId;
    }

    public String getCategoryTagName() {
        return categoryTagName;
    }

    public String getCurrency() {
        return currency;
    }

    public long getNetMinor() {
        return netMinor;
    }
}

File: pocketr-api/src/main/kotlin/com/decrux/pocketr/api/repositories/projections/MonthlyExpenseProjection.kt
package com.decrux.pocketr.api.repositories.projections

import com.decrux.pocketr.api.entities.db.ledger.AccountStatus
import java.util.UUID

data class MonthlyExpenseProjection(
    val expenseAccountId: UUID,
    val expenseAccountName: String,
    val categoryTagId: UUID?,
    val categoryTagName: String?,
    val currency: String,
    val netMinor: Long,
    val categoryTagColor: String? = null,
    val expenseAccountStatus: AccountStatus = AccountStatus.ACTIVE,
)
